python/tvm/relay/backend/contrib/tidl.py

```diff
@@ -0,0 +1,1403 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""TIDL backend compiler"""
+import os
+import sys
+import subprocess
+
+import numpy as np
+import tvm
+from tvm import relay
+import topi
+from topi.util import get_const_tuple
+import ctypes
+import re
+
+from tvm.relay.expr_functor import ExprMutator, ExprVisitor
+from tvm.relay.expr import Call, Constant, Tuple, GlobalVar
+from tvm.relay.function import Function
+from tvm.relay import transform
+from tvm.relay.build_module import bind_params_by_name
+from tvm.contrib import graph_runtime
+from tvm.relay.op.contrib import tidl
+
+if os.getenv("TIDL_TOOLS_PATH") is None:
+    sys.exit("Environment variable TIDL_TOOLS_PATH is not set!")
+else:
+    tidl_tools_path = os.getenv("TIDL_TOOLS_PATH")
+tidl_calib_tool = os.path.join(tidl_tools_path, "eve_test_dl_algo_ref.out")
+tidl_import_lib = os.path.join(tidl_tools_path, "tidl_relayImport.so")
+_tidl_mod = ctypes.CDLL(tidl_import_lib, mode=ctypes.RTLD_GLOBAL)

```

Comment:
Can this setup be moved, perhaps to `TIDLCompiler.__init__`? `TIDL_TOOLS_PATH` can be an argument to the constructor instead of an environment variable.

Reply:
Done.